sky/data/mounting_utils.py

```diff
@@ -13,6 +14,9 @@
 _RENAME_DIR_LIMIT = 10000
 # https://github.com/GoogleCloudPlatform/gcsfuse/releases
 GCSFUSE_VERSION = '2.2.0'
+# https://github.com/Azure/azure-storage-fuse/releases
+BLOBFUSE2_VERSION = '2.2.0'
+_BLOBFUSE_CACHE_DIR = '~/.sky/blobfuse2_cache'
```

Comment:
@landscapepainter found what I think is a critical bug - the same cache is used across *all* mounts, meaning files with the same name and relative path will conflict.

Reply:
Thanks for catching this @visatish. It has been address at [89645ed](https://github.com/skypilot-org/skypilot/pull/3032/commits/89645ed309e76d2d660a301e128f883e309802f6).